lib/features/scan_nfc/nfc_scan_screen.dart

```diff
@@ -1,55 +1,132 @@
+import 'dart:io';
+
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:givt_app_kids/core/app/flows.dart';
+import 'package:givt_app_kids/core/app/pages.dart';
 import 'package:givt_app_kids/features/coin_flow/cubit/search_coin_cubit.dart';
 import 'package:givt_app_kids/features/coin_flow/widgets/coin_found.dart';
+import 'package:givt_app_kids/features/giving_flow/organisation_details/cubit/organisation_details_cubit.dart';
+import 'package:givt_app_kids/features/scan_nfc/widgets/abdroid_nfc_found_bottomsheet.dart';
+import 'package:givt_app_kids/features/scan_nfc/widgets/android_nfc_scanning_bottomsheet.dart';
 import 'package:givt_app_kids/features/coin_flow/widgets/search_coin_animated_widget.dart';
 import 'package:givt_app_kids/features/scan_nfc/cubit/scan_nfc_cubit.dart';
 import 'package:givt_app_kids/features/scan_nfc/widgets/start_scan_nfc_button.dart';
+import 'package:givt_app_kids/shared/widgets/floating_action_button.dart'
+    as custom;
 import 'package:givt_app_kids/shared/widgets/givt_back_button.dart';
+import 'package:go_router/go_router.dart';
 
 class NFCScanPage extends StatelessWidget {
   const NFCScanPage({super.key});
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        backgroundColor: Colors.transparent,
-        elevation: 0,
-        leading: const GivtBackButton(),
-      ),
-      body: BlocBuilder<ScanNfcCubit, ScanNfcState>(
-        builder: (context, state) {
-          return Center(
+    return BlocConsumer<ScanNfcCubit, ScanNfcState>(
+      listener: (context, state) {
+        final scanNfcCubit = context.read<ScanNfcCubit>();
+        if (state.scanNFCStatus == ScanNFCStatus.scanning &&
+            Platform.isAndroid) {
+          showModalBottomSheet<void>(
+            context: context,
+            shape: RoundedRectangleBorder(
+              borderRadius: BorderRadius.circular(10.0),
+            ),
+            builder: (BuildContext context) {
+              return BlocProvider.value(
+                value: scanNfcCubit,
+                child: Padding(
+                  padding: const EdgeInsets.all(20),
+                  child: BlocBuilder<ScanNfcCubit, ScanNfcState>(
+                    builder: (context, state) {
+                      if (state.scanNFCStatus == ScanNFCStatus.scanned) {
```

Comment:
on line 28 its status **scanning**, can it be here status **scanned**? 🤔

Reply:
BottomSheet creates a new context, thats why I need a new bloc provired and bloc builder. And it is ScanNFCStatus.scanned, because the bottomsheet updates when the state changes from scanning to scanned, without popping. This is basically the animation stopping.